bedrock/products/tests/test_views.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+
+from django.test.client import RequestFactory
+
+from mock import patch
+
+from bedrock.products import views
+from bedrock.mozorg.tests import TestCase
+
+
+class TestVPNInviteWaitlist(TestCase):
+    def setUp(self):
+        patcher = patch('bedrock.products.views.basket.subscribe')
+        self.mock_subscribe = patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def _request(self, data, expected_status=200, locale='en-US'):
+        req = RequestFactory().post('/', data)
+        req.locale = locale
+        resp = views.vpn_invite_waitlist(req)
+        assert resp.status_code == expected_status
+        return json.loads(resp.content)
+
+    def test_form_success(self):
+        resp_data = self._request({
+            'newsletters': 'guardian-vpn-waitlist',
+            'email': 'test@example.com',
+            'country': 'us',
+            'privacy': True,
+            'fmt': 'H',
+            'lang': ''
```

Comment:
@pmac thanks again for helping out yesterday. I tried to put together some basic tests for the new view earlier this morning, and what I have here seems to be working ok. But one strange thing I couldn't quite figure out is why the tests here won't accept any value for `lang`, other than an empty string? If I try and pass `en`, which is the default used in the page, then the tests say it's not an accepted value. Any idea where I'm going wrong here? 

(also: not urgent, thanks)

Reply:
I'm guessing it has to do with how the form populates its language choices. It uses basket and product-details data to provide the choices list. If during the tests these data aren't available I'm guessing the list of choices will be empty. Are you running these locally? If so maybe run `bin/run-db-download.py --force` to get a fresh DB and see if that makes a difference. When I do that and then run the following I do get `en` as a choice:

```python
>>> from bedrock.newsletter.forms import get_lang_choices
>>> get_lang_choices('guardian-vpn-waitlist')
[['en', 'English']]
```

Reply:
Also nice work on these tests! 🥇 for tests!!

Reply:
Ah, looks like the tests don't have access to this when they run. I solved it with:

```
@patch('bedrock.newsletter.forms.get_lang_choices', lambda *x: [['en', 'English']])
```

Thanks!